core/lib/Cardano/Address/Script.hs

```diff
@@ -362,11 +362,13 @@ recommendedValidation = \case
         when (L.null (omitTimelocks script)) $ Left EmptyList
         when (hasDuplicate script) $ Left DuplicateSignatures
         when (redundantTimelocks script) $ Left RedundantTimelocks
+        when (timelockTrap script) $ Left TimelockTrap
```

Comment:
What does TimelockTrap mean?
Is it basically that the interval ends before it starts?

Reply:
it it the situation when we have script ALL with timelocks. if we have for example all [active_until 10, active_after 20, something else] then in order to have this section satisfied all elements are to be satisfied. But it is not possible to be BOTH before slot=10 and after slot=20 at the same time! So we DO NOT RECOMMEND it in validation as it means the script will lock spending from funds or staking from them. Although ledger would accept this script, and hence user can, on his own responsibility use required validation, and pursue with this script